### PR TITLE
feat(localstorev2): migration

### PR DESCRIPTION
**Pull request by vladopajic**
Migration part of #3750 issue.

This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/3791)

### CODE REVIEW

File: pkg/localstorev2/migration/step_01_test.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+	localmigration "github.com/ethersphere/bee/pkg/localstorev2/migration"
+	"github.com/ethersphere/bee/pkg/storagev2/inmemstore"
+)
+
+func Test_Step_01(t *testing.T) {
+	t.Parallel()
+
+	stepFn := localmigration.Step_01
+	dummyObj := localmigration.NewDummyObj(1635)
```
Comment (aloknerurkar) [inline pkg/localstorev2/migration/step_01_test.go:20]: Again here you could just check version entry was created.

File: pkg/localstorev2/migration/step_01.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+	storage "github.com/ethersphere/bee/pkg/storagev2"
+)
+
+// step_01 serves as example for setting up migration step.
+//
+// In this example migration step will ensure that dummyObj with id:1635 is
+// defined in the store.
+func step_01(s storage.Store) error {
+	obj := newDummyObj(1635)
```
[the remark this answers is not in the document]
Comment (vladopajic) [reply pkg/localstorev2/migration/step_01.go:20]: Migration step function is only making changes to the `Store` in logic that is defined by this function. 
Version index is changed only when migration is performed on the `Store` by calling ` migration.Migrate(...)`

If `step_01` is not performing any changes to the store, then it's tests case would be to only ensure that state is note changed after this step function is executed.

[the remark this answers is not in the document]
Comment (vladopajic) [reply pkg/localstorev2/migration/step_01.go]: New commit was pushed were dummy object is removed in favor for empty step function.

File: pkg/localstorev2/migration/step_01_test.go (hunk 1 above)
Comment (vladopajic) [reply pkg/localstorev2/migration/step_01_test.go]: Please, read explanation in comment above.

**Review by aloknerurkar — APPROVED** [review]

**Pull request merged**